server/server.go

```diff
@@ -75,8 +75,7 @@ func NewServer(ctx *Context, stopper *util.Stopper) (*Server, error) {
 		return nil, util.Error("ctx must not be null")
 	}
 
-	// If the specified address includes no host component, use the hostname.
-	addr := util.EnsureHost(ctx.Addr)
+	addr := ctx.Addr
```

Comment:
This is all well and good but if you pass in :8080 to the rpc server, what does it report from `Addr()` after binding to all network interfaces? That address is what we gossip with the node descriptor so that other nodes can connect.